### PR TITLE
JP-3593: 2nd group saturation part 2

### PR DESCRIPTION
**Pull request by drlaw1558**
Ready for review: this addresses JP-3593, flagging cosmic ray saturation in group 2 data in a consistent manner between both IRS2 and non-IRS2 modes using the new use_readpatt keyword.  This PR is tied to JWST PR https://github.com/spacetelescope/jwst/pull/8731

**Checklist**

- [ ] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [ ] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [ ] added relevant label(s)

### CODE REVIEW

**Comment by drlaw1558** [comment]
@tapastro Should now be ready for review.

**Review by kmacdonald-stsci — APPROVED** [review]

**Review by schlafly — APPROVED** [review]
Thanks, this looks like a good algorithm to me.  Thank you for making it handle ~arbitrary read patterns!

File: CHANGES.rst [hunk 1]
```diff
@@ -21,6 +21,12 @@ ramp_fitting
 
 - Fixed memory leak in C-extension.[#281]
 
+saturation
+~~~~~~~~~~
+
+- Add option for using the readout pattern information to improve saturation flagging
+  in grouped data. [#283]
+
```
Comment (zacharyburnett) [inline CHANGES.rst:29]:
```suggestion
```
#284 switched change log handling to `towncrier`, so this change log entry should be a file in `changes/`:
```shell
echo "[saturation] Add option for using the readout pattern information to improve saturation flagging in grouped data." > changes/283.bugfix.rst
```

(new PRs will have these instructions in the checklist)

Comment (drlaw1558) [reply CHANGES.rst]: Done

**Comment by tapastro** [comment]
Running a set of regression tests on the PR pair here: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1722/

**Review by tapastro — APPROVED** [review]

**Pull request merged**